data_management/prov.py

```diff
@@ -1,133 +1,356 @@
+                file_id,
+                (
+                    (PROV_TYPE, 'file'),
+                    *_generate_object_meta(obj),
+                ),
+            )
 
+            # add external object linked to the data product
+            _add_external_object(doc, data_product, file_entity)
 
-def generate_prov_document(code_run):
+            _add_author_agents(obj.authors.all(), doc, file_entity)
+
+            if input_objects:
+                # add link to the code run
+                doc.used(
+                    cr_activity, file_entity, None, None, {PROV_ROLE: 'input data'})
+                # add the link to the data product
+                doc.wasDerivedFrom(dp_entity, file_entity)
+            else:
+                # add the link to the code run
+                doc.wasGeneratedBy(file_entity, cr_activity)
+
+
+def _add_model_config(cr_activity, doc, model_config):
+    """
+    Add model config to the code run activity.
+
+    @param cr_activity: a prov.activity representing the code run
+    @param doc: a ProvDocument that the entities will belong to
+    @param model_config: a model_config object
+
+    """
+    model_config_entity = doc.entity(
+        f'api/object/{model_config.id}', (*_generate_object_meta(model_config),)
+    )
+
+    _add_author_agents(model_config.authors.all(), doc, model_config_entity)
+    doc.used(
+        cr_activity, model_config_entity, None, None, {PROV_ROLE: 'model configuration'}
+    )
+
+
+def _add_submission_script(cr_activity, doc, submission_script):
+    """
+    Add submission script to the code run activity.
+
+    @param cr_activity: a prov.activity representing the code run
+    @param doc: a ProvDocument that the entities will belong to
+    @param submission_script: a submission_script object
+
+    """
+    submission_script_entity = doc.entity(
+        'api/object/' + str(submission_script.id),
+        (*_generate_object_meta(submission_script),),
+    )
+
+    _add_author_agents(submission_script.authors.all(), doc, submission_script_entity)
+    doc.used(
+        cr_activity,
+        submission_script_entity,
+        None,
+        None,
+        {PROV_ROLE: 'submission script'},
+    )
+
+
+def get_whole_object_component(components):
+    for component in components:
+        if component.whole_object:
+            return component
+
+
+def generate_prov_document(data_product):
     """
-    Generate a PROV document for a CodeRun detailing all the input and outputs and how they were generated.
+    Generate a PROV document for a DataProduct detailing all the input and outputs and
+    how they were generated.
 
     This uses the W3C PROV ontology (https://www.w3.org/TR/prov-o/).
 
-    :param code_run: The CodeRun to generate the PROV document for
+    :param data_product: The DataProduct to generate the PROV document for
+
     :return: A PROV-O document
+
     """
+    url = ''.join(['http://', get_current_site(None).domain, '/'])
```

Comment:
This isn't correct in general, e.g. for the remote registry we use `https`. Also, when testing this branch in a local registry it gives the wrong site name (for me at least):
```
    "prefix": {
        "default": "http://example.com/"
    },
```
One way to fix this problem, which I've tested, is to replace this line with:
```
url = request.build_absolute_uri('/')
```
 For this to work, in the `get` method in `ProvReportView` (in `data_management/rest/views.py`) you need to pass `request` into `generate_prov_document`.